torch/csrc/jit/type.h

```diff
@@ -452,6 +453,54 @@ struct TORCH_API ListType : public Type {
   bool has_free_variables_;
 };
 
+struct FutureType;
+using FutureTypePtr = std::shared_ptr<FutureType>;
+
+struct TORCH_API FutureType : public Type {
+  // It's not exactly a singleton, but there should be exactly once instance of
+  // Future[T] for every T
+  static constexpr bool is_singleton = true;
+  friend struct Type;
+  template<typename ... T>
+  static FutureTypePtr create( T&& ... all) {
+    return FutureTypePtr(new FutureType( std::forward<T>(all)... )); // NOLINT(modernize-make-shared)
```

Comment:
nit: I know this what the other classes do, but I don't want to propagate it. It produces bad errors. Just write `create(TypePtr elem)`